### ISSUE TITLE
[Android] Autofill is not working

### ISSUE THREAD

**Issue by Nike682631** [flutter/flutter#97633]
## Steps to Reproduce

The following code doesn't show any autofill hints in my physical Android device nor on web

**Expected results:** Autofill credentials popping up when textfield is being focused

**Actual results:** Nothing comes up

<details>
<summary>Code sample</summary>

```dart
                                              Form(
                                                key: _loginFormKey,
                                                child: AutofillGroup(
                                                  child: Column(
                                                    children: <Widget>[
                                                      Padding(
                                                        padding: const EdgeInsets.only(bottom: kDefaultPadding),
                                                        child: TextField(
                                                          key: _usernameKey,
                                                          autofillHints: const <String>[AutofillHints.username],
                                                          keyboardType: TextInputType.emailAddress,
                                                          decoration: InputDecoration(
                                                            suffixIcon: const Icon(Icons.person_outline),
                                                            hintStyle: _kbt.sb2().copyWith(color: Colors.grey.withOpacity(0.5)),
                                                            contentPadding: const EdgeInsets.symmetric(horizontal: 5.0),
                                                            labelText: 'E-Mail',
                                                          ),
                                                          controller: _usernameController,
                                                          autocorrect: false,
                                                          onSubmitted: (String value) {
                                                            if (state is! LoginLoadInProgress || state is! LoginLoadSuccess) {
                                                              _onLoginButtonPressed();
                                                            }
                                                          },
                                                        ),
                                                      ),
                                                      TextField(
                                                        key: _passwordKey,
                                                        autofillHints: const <String>[AutofillHints.password],
                                                        decoration: InputDecoration(
                                                          suffixIcon: const Icon(Icons.lock_outline),
                                                          hintStyle: _kbt.sb2().copyWith(color: Colors.grey.withOpacity(0.5)),
                                                          contentPadding: const EdgeInsets.symmetric(horizontal: 5.0),
                                                          labelText: 'general.password'.tr(),
                                                        ),
                                                        autocorrect: false,
                                                        controller: _passwordController,
                                                        obscureText: true,
                                                        onEditingComplete: () => TextInput.finishAutofillContext(),
                                                        onSubmitted: (String value) {
                                                          if (state is! LoginLoadInProgress || state is! LoginLoadSuccess) {
                                                            _onLoginButtonPressed();
                                                          }
                                                        },
                                                      ),
                                                      InkWell(
                                                        hoverColor: Colors.transparent,
                                                        splashColor: Colors.transparent,
                                                        highlightColor: Colors.transparent,
                                                        onTap: () {
                                                          setState(() {
                                                            _remember = !_remember;
                                                          });

                                                          if (_remember) _showRememberInfo();
                                                        },
                                                        child: Row(
                                                          mainAxisAlignment: MainAxisAlignment.center,
                                                          children: <Widget>[
                                                            Checkbox(
                                                              checkColor: Colors.white,
                                                              activeColor: Theme.of(context).colorScheme.primary,
                                                              value: _remember,
                                                              onChanged: (bool? remember) {
                                                                setState(() {
                                                                  _remember = remember!;
                                                                });

                                                                if (_remember) _showRememberInfo();
                                                              },
                                                            ),
                                                            Flexible(
                                                              child: _kbt.h6('account.stayLoggedIn'),
                                                            ),
                                                          ],
                                                        ),
                                                      ),
                                                    ],
                                                  ),
                                                ),
                                              ),
```
</details>

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel stable, 2.8.1, on Manjaro Linux
    5.15.7-1-rt23-MANJARO, locale en_IN)
    • Flutter version 2.8.1 at /home/nikunj/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (7 weeks ago), 2021-12-16 08:37:33
      -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android SDK version
    31.0.0)
    • Android SDK at /home/nikunj/Android/Sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at:
      /var/lib/snapd/snap/android-studio/115/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at
      /var/lib/snapd/snap/android-studio/115/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      11.0.10+0-b96-7249189)

[✓] VS Code
    • VS Code at /snap/code/current
    • Flutter extension version 3.34.0

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome
      96.0.4664.110

• No issues found!

```

</details>
I have also referred to: https://github.com/flutter/flutter/issues/57320 but it did not prove helpful to me.

**Comment by maheshmnj** [comment]
Hi @Nike682631, Thanks for filing the issue. This was a known issue on `stable 2.8.0` and I can confirm that this is fixed on the master channel. It was fixed here https://github.com/flutter/flutter/pull/95210.

You may switch to the latest master by running 

```
flutter channel master
flutter upgrade
```

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 2.10.0-1.0.pre.371, on macOS 12.1 21C52 darwin-arm, locale en-GB)
    • Flutter version 2.10.0-1.0.pre.371 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d1285eef1b (6 hours ago), 2022-02-02 00:50:20 -0500
    • Engine revision 5bbdb7a6f7
    • Dart version 2.17.0 (build 2.17.0-74.0.dev)
    • DevTools version 2.10.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • Redmi K20 Pro (mobile) • 192.168.1.2:5555 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)        • macos            • darwin-arm64   • macOS 12.1 21C52 darwin-arm
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 97.0.4692.99

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>


 <details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: AutofillGroup(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            const TextField(
              autofillHints: [AutofillHints.email],
            ),
            const SizedBox(
              height: 20,
            ),
            const TextField(
              autofillHints: [AutofillHints.password],
            ),
            const SizedBox(
              height: 20,
            ),
            ElevatedButton(
              child: const Text("Save"),
              onPressed: () {
                TextInput.finishAutofillContext(shouldSave: true);
                debugPrint("Saving");
              },
            ),
          ],
        ),
      ),
    );
  }
}

```
</details>

The web-specific issue is being tracked here https://github.com/flutter/flutter/issues/87735

**Comment by ugurberkecan** [comment]
Hi @maheshmnj password fill does not work.

**Comment by maheshmnj** [comment]
@ugurberkecan, could you please check on the latest master channel as mentioned above

**Comment by ugurberkecan** [comment]
Still doesnt work on chrome and safari.

It is only work on mozilla.

**Comment by maheshmnj** [comment]
Autofill for Android doesn't work on stable 2.10.0 can this be cherry picked?
cc: @zanderso,

**Comment by zanderso** [comment]
This looks like a duplicate of https://github.com/flutter/flutter/issues/94043. The fix is planned to go out in the first 2.10 hotfix release.
